### ISSUE TITLE
How to use two target values for in the loss function?

### ISSUE THREAD

**Issue by danijar** [fchollet/keras#3238]
I'm implementing DQN which has an output layer of the size of available actions. For each example, only the output node corresponding to the taken action is trained towards a target. So I have a custom loss function like this:

``` python
def loss(target, pred, action):
    mse = tf.reduce_mean((pred - target) ** 2, 0)
    masked = mse * tf.one_hot(action)
    return tf.reduce_sum(masked)
```

How can I use two values in addition to the prediction? If possible, it would be nice to pass in the action during training only, not during prediction where all outputs are of interest.

**Comment by saicoco** [comment]
Maybe you can do this:   
```` python
model.compile(optimizer=sgd, loss=loss, metrics=[loss])
model.fit(train_x, [target, action], batch_size=batch_size, nb_epoch=epoches...)
```  
here `loss` is your function, and there are two ouputs of your model. When I train face and audio model with the same label, I just do in this way:
```
model.fit([train_face, train_audio], [train_label, train_label], batch_size=batch_size, nb_epoch=epoches...)
```
try it, I think it will work.

**Comment by danijar** [comment]
Hi @saicoco, I moved away from Keras because I felt it was not flexible enough for what I'm doing. Thanks for the answer, anyway.

**Comment by saicoco** [comment]
@danijar There are many implements that are value to learn,  you can refer them when you use `theano` to code.

**Comment by danijar** [comment]
Yes! :) Should we close this issue or keep it open in case someone else is interested in the question?

**Comment by saicoco** [comment]
Open it just, there maybe someone confuse this.
